### DISCUSSION TITLE
Unresolved reference: MapViewPosition

### DISCUSSION THREAD

**Discussion by Expolevin** [mapsforge/mapsforge#1403] [Q&A] [answered]
Hi I'm new to offline map development and I have a problem with MapViewPosition, and import it into the main file And the MapViewPosition comes out in red letters I also put its dependence on build.gradle and I still keep throwing the error of: "Unresolved reference: MapViewPosition" I need help in that please.

dependencies {

    implementation 'androidx.core:core-ktx:1.9.0'
    implementation 'androidx.appcompat:appcompat:1.6.1'
    implementation 'com.google.android.material:material:1.8.0'
    implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
    testImplementation 'junit:junit:4.13.2'
    androidTestImplementation 'androidx.test.ext:junit:1.1.5'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
    //---

    implementation 'org.mapsforge:mapsforge-map-android:0.18.0'
    implementation 'org.mapsforge:mapsforge-core:0.18.0'
    implementation 'org.mapsforge:mapsforge-map:0.18.0'
    implementation 'org.mapsforge:mapsforge-map-reader:0.18.0'
    implementation 'org.mapsforge:mapsforge-themes:0.18.0'
    implementation 'net.sf.kxml:kxml2:2.3.0'
    implementation 'org.mapsforge:mapsforge-map-android:0.18.0'
    implementation 'com.caverock:androidsvg:1.4'
    implementation 'org.mapsforge:mapsforge-core:0.18.0'
    implementation 'org.mapsforge:mapsforge-poi:0.18.0'
    implementation 'org.mapsforge:mapsforge-poi-android:0.18.0'
    implementation 'org.mapsforge:sqlite-android:0.18.0:natives-armeabi-v7a'
    implementation 'org.mapsforge:sqlite-android:0.18.0:natives-arm64-v8a'
    implementation 'org.mapsforge:sqlite-android:0.18.0:natives-x86'
    implementation 'org.mapsforge:sqlite-android:0.18.0:natives-x86_64'
    implementation 'org.mapsforge:sqlite-android:0.18.0'
    implementation 'org.mapsforge:mapsforge-map-awt:0.18.0'
    implementation 'com.formdev:svgSalamander:1.1.3'
    implementation 'com.google.android.gms:play-services-location:21.0.1'
    implementation 'org.osmdroid:osmdroid-android:6.1.11'
    

}
///EN EL MAIN//
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import com.example.prueba.databinding.ActivityMainBinding
import org.mapsforge.core.model.LatLong
import org.mapsforge.map.android.graphics.AndroidGraphicFactory
import org.mapsforge.map.android.util.AndroidUtil
import org.mapsforge.map.layer.Layer
import org.mapsforge.map.layer.renderer.TileRendererLayer
import org.mapsforge.map.reader.MapFile
import org.mapsforge.map.rendertheme.InternalRenderTheme
import java.io.FileInputStream
import org.mapsforge.map.android.view.MapViewPosition

**Comment by devemux86** [comment, marked as answer]
For Android development, you need these dependencies:
- [Core](https://github.com/mapsforge/mapsforge/blob/master/docs/Integration.md#core)
- [Android](https://github.com/mapsforge/mapsforge/blob/master/docs/Integration.md#android)

See also the [integration guide](https://github.com/mapsforge/mapsforge/blob/master/docs/Integration.md).
